File: web_shop_backend/src/main/java/com/lap/web_shop_backend/model/Shipment.java
package com.lap.web_shop_backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.sql.Date;
import java.sql.Timestamp;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Shipment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "shipping_address")
    private Address shipping_address;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @CreationTimestamp
    private LocalDateTime shipment_date;

    @OneToMany(mappedBy = "shipment", cascade = CascadeType.ALL)
    private List<Shipment_product> shipmentProducts;
}

File: web_shop_frontend/src/models/Shipment.ts
import Address from './Address'
import Shipment_product from './Shipment_product'
import Customer from './Customer'

export default class Shipment {
  id: Number
  shipmentDate: Date
  shipmentProducts: Shipment_product[]
  shipping_address: Address
  customer: Customer

  constructor(data: any) {
    this.id = data.id
    this.shipmentDate = data.date
    this.shipping_address = new Address(data.shipping_address)
    this.customer = new Customer(data.customer)

    this.shipmentProducts = []
    for (const shipment_product of data.shipmentProducts)
      this.shipmentProducts.push(new Shipment_product(shipment_product))
  }
}
